### DISCUSSION TITLE
What prediction length for one step-ahead forecast?

### DISCUSSION THREAD

**Discussion by mtomic123** [awslabs/gluonts#2647] [Q&A]
I would like to build a `DeepVAR` model that predicts only one-step ahead. Given my hourly data, this means I would only like to predict the next hour.

I am not sure how `prediction_length` works exactly and whether it is advisable to use `prediction_length=1` or to build a model with a larger prediction_length but just use the first prediction.

I see three different options for my use case:

1. `prediction_length=1`
2. `prediction_length=1` but context_length > 1
3.  larger number such as `prediction_length = 24` , and take only first prediction

Any insights?

**Comment by jaheba** [comment]
If you want to predict just one step ahead, `prediction_length` should be `1`.

I would assume that the results wouldn't differ much compared to `prediction_length = 24`, but notice that in that case the model tries to optimise for `24` values instead of the one that you are interested in.

I would set `context_length` to a frequency dependent value, in your case maybe to `12` or `24`.
